src/hooks/useCheckImage.ts

```diff
@@ -0,0 +1,27 @@
+import { useEffect, useState } from "react";
+
+export const useCheckImage = (
+  src: string
+): [boolean, boolean, string | undefined] => {
+  const [isLoading, setLoading] = useState(false);
+  const [error, setError] = useState<string>();
+
+  useEffect(() => {
```

Comment:
Can you use `useAsync` here instead? It provides both error and loading out of the box and you won't need the 2 states above.

https://github.com/streamich/react-use/blob/master/docs/useAsync.md

Reply:
Great suggestion! Fixed in https://github.com/sparkletown/sparkle/pull/1995/commits/c2dfda2754be5b6a58f54916649a3aa6da4951be